src/lib/build-server.ts

```diff
@@ -42,44 +48,36 @@ export async function buildServer({
 		logger: envToLogger[NODE_ENV] ?? true,
 		disableRequestLogging: !(NODE_ENV === "development" || NODE_ENV === "test"),
 	});
-	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-	// @ts-expect-error
-	await server.register(fastifySwagger, {
-		mode: "dynamic",
-		openapi: {
-			info: {
-				title: String,
-				description: String,
-				version: String,
-			},
-			externalDocs: Object,
-
-			components: Object,
-		},
-	});
+	server.register(cors, corsConfiguration);
 
+	server.register(fastifySwagger, swaggerOptions);
 	// Set rate limit
-	await server.register(import("@fastify/rate-limit"), {
+	server.register(import("@fastify/rate-limit"), {
 		max: 30,
 		timeWindow: "1 minute",
 	});
-
-	registerDocumentationRoute(server);
-	registerRootRoute(server);
-	registerHealthRoute(server);
-	registerCountDocumentsRoute(server);
+	server.register(fastifySwaggerUi, documentationRouteOptions);
+	server.register(rootRoute);
+	server.register(healthRoute, { prefix: "/health" });
+	// FIXME: We register at /processed_documents and add one handler for /count. Why?
+	server.register(countDocumentsRoute, {
+		prefix: "/processed_documents",
+	});
 	server.register(feedbackRoute, { prefix: "/feedbacks" });
-
-	registerSearchDocumentsRoute(
-		server,
+	server.register(searchDocumentsRoute, {
+		prefix: "/vector-search",
 		OPENAI_KEY,
 		OPENAI_EMBEDDING_MODEL,
 		OPENAI_MODEL,
-	);
-	registerGenerateAnswerRoute(server, OPENAI_MODEL, OPENAI_KEY);
-	registerLoadUserRequestRoute(server);
+	});
+	server.register(generateAnswerRoute, {
+		prefix: "/generate-answer",
+		OPENAI_MODEL,
+		OPENAI_KEY,
+	});
+	server.register(loadUserRequestRoute, { prefix: "/requests" });
 
-	server.setErrorHandler(function (error, request, reply) {
+	-server.setErrorHandler(function (error, request, reply) {
```

Comment:
Just for my understanding: What does `-server` mean?

Reply:
ups. This is an error. I wounder why this still works and does not fail. JS magic

Reply:
oh wow. I thought it was some magic operator for error handling :D